### DISCUSSION TITLE
How to use chakra factory with custom react component?

### DISCUSSION THREAD

**Discussion by AhsanSarwar45** [chakra-ui/chakra-ui#6960] [Q&A]
I have a react component `UnderlinedButton` that accepts `React.CSSProperties` in addition to some other props. Now I want to use that component with Chakra theme-aware style props. As I understood, chakra factory is the way to do it. But I can't seem to get it to work.

My code is basically like this:
```jsx
const UnderlinedButton = (props: UnderlinedButtonProps) => {
  const {className, isUnderlined, ...cssProps} = props;
  return(
    <motion.div
      className={className}
      style={...cssProps}
    >
      ...
    </motion.div>
    )
}

const ChakraUnderlinedButton = chakra(UnderlinedButton);
```
I use my custom component like so:
```jsx
<UnderlinedButton color="red">Hello World</UnderlinedButton>
```
But when I use the chakra enabled version, the theme props don't get passed to my component.
```jsx
<ChakraUnderlinedButton color="red.500">
  Hello World
</ChakraUnderlinedButton>
```
What am I doing wrong here?

**Comment by HaleyWardo** [comment]
The motion div is what you want to be a chakra wrapped component since it will be the one getting the styles applied. So you'll want to do something like this:

``` 
const MotionDiv = chakra(motion.div);
``` 

Then replace it inside of your component.

```
const UnderlinedButton = (props: UnderlinedButtonProps) => {
  const {className, isUnderlined, ...cssProps} = props;
  return(
    <MotionDiv
      className={className}
      style={...cssProps}
    >
      ...
    </MotionDiv>
    )
}
```

With doing this, you don't need to wrap `UnderlinedButton` in the chakra function. 

So the full implementation should look like this:

```
const MotionDiv = chakra(motion.div);

const UnderlinedButton = (props: UnderlinedButtonProps) => {
  const {className, isUnderlined, ...cssProps} = props;
  return(
    <MotionDiv
      className={className}
      style={...cssProps}
    >
      ...
    </MotionDiv>
    )
}

<UnderlinedButton color="red">Hello World</UnderlinedButton>

```
